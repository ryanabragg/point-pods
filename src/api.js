import localforage from 'localforage';
import React from 'react';

const api = {
  store: localforage.createInstance({
    name: 'Point Pods',
    storeName: 'point_pods',
    description: 'some description'
  }),
};

export default api;
export const APIContext = React.createContext(api);
export function withAPI(Component) {
  return function APIComponent(props) {
    return (
      <APIContext.Consumer>
        {context => <Component {...props} api={context} />}
      </APIContext.Consumer>
    );
  };
}
