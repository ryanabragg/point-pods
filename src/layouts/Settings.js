import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import AppMenu from '../components/AppMenu';
import AppSettings from '../components/AppSettings';

const styles = theme => ({});

class Settings extends Component {
  goBack = () => {
    this.props.history.go(-1);
  };

  render() {
    return (
      <div>
        <AppMenu title='Settings' usesBackIcon onBack={this.goBack} />
        <AppSettings />
      </div>
    );
  }
}

Settings.defaultProps = {
  history: {
    push: () => null,
    replace: () => null,
  },
};

Settings.propTypes = {
  classes: PropTypes.object.isRequired, // added by withStyles
  history: PropTypes.object, // added by parent
};

export default withStyles(styles, { withTheme: true })(Settings);
