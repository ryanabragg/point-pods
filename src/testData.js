import api from './api';

const testSettings = {
  pairingMethod: api.pairingMethods.HISTORIC,
  pairingMethodInitial: api.pairingMethods.HISTORIC,
  podSizeMinimum: 4,
  podSizeMaximum: 7,
};

const testPlayers = [{
  id: 'mock0',
  name: 'test0',
  points: 0,
}, {
  id: 'mock1',
  name: 'test1',
  points: 1,
}, {
  id: 'mock2',
  name: 'test2',
  points: 2,
}, {
  id: 'mock3',
  name: 'test3',
  points: 3,
}, {
  id: 'mock4',
  name: 'test4',
  points: 4,
}, {
  id: 'mock5',
  name: 'test5',
  points: 5,
}, {
  id: 'mock6',
  name: 'test6',
  points: 6,
}, {
  id: 'mock7',
  name: 'test7',
  points: 7,
}, {
  id: 'mock8',
  name: 'test8',
  points: 8,
}, {
  id: 'mock9',
  name: 'test9',
  points: 9,
}];

const testTournaments = [{
  id: 'mock0',
  name: 'test0',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Etiam urna tortor, laoreet vitae commodo porttitor, mattis id purus.',
  date: new Date('2018-01-01'),
  done: false,
  staging: true,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 3,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 3 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 0,
    participated: false,
    dropped: false,
  }],
}, {
  id: 'mock1',
  name: 'test1',
  category: 'cat1',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed molestie suscipit suscipit. Nullam sollicitudin tempus pretium.',
  date: new Date('2018-02-01'),
  done: true,
  staging: false,
  pairingMethod: api.pairingMethods.POINTS,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 5,
  rounds: 2,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 0 },
    2: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 1 },
    2: { pod: 2, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
    2: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 3 },
    2: { pod: 2, points: 3 },
  }, {
    id: 'mock4',
    name: 'test4',
    points: 4,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 4 },
    2: { pod: 1, points: 4 },
  }, {
    id: 'mock5',
    name: 'test5',
    points: 5,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 5 },
    2: { pod: 2, points: 5 },
  }, {
    id: 'mock6',
    name: 'test6',
    points: 6,
    participated: true,
    dropped: false,
    2: { pod: 2, points: 6 },
  }],
}, {
  id: 'mock2',
  name: 'test2',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Sed ullamcorper metus vitae nisi rhoncus placerat.',
  date: new Date('2018-01-01'),
  done: false,
  staging: false,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 3 },
  }],
}];

const mockRoute = {
  history: {
    push: jest.fn(to => to),
    replace: jest.fn(to => to),
  },
  match: {
    params: {},
  },
  matchWithInvalid: {
    params: {
      id: 'invalid',
    },
  },
  matchWithID0: {
    params: {
      id: 'mock0',
    },
  },
  matchWithID1: {
    params: {
      id: 'mock1',
    },
  },
};

let mockStore = {};
const mockStorage = {
  setItem: jest.fn((key, val) => Promise.resolve((mockStore[key] = val))),
  getItem: jest.fn(key => Promise.resolve(mockStore[key])),
  removeItem: jest.fn(key => { delete mockStore[key]; Promise.resolve(); }),
  clear: jest.fn(() => Promise.resolve((mockStore = {}))),
  keys: jest.fn(() => Promise.resolve(Object.keys(mockStore))),
  length: jest.fn(() => Promise.resolve(Object.keys(mockStore).length)),
};

let defaultStore = null;
function mockAPI(api, storage) {
  defaultStore = api.store;
  api.store = storage;
  mockStore = {
    settings: testSettings,
    players: testPlayers,
    tournaments: testTournaments,
  };
}
function demockAPI(api) {
  api.store = defaultStore;
  defaultStore = null;
}

export {
  testSettings,
  testPlayers,
  testTournaments,
  mockRoute,
  mockStore,
  mockStorage,
  mockAPI,
  demockAPI,
};
