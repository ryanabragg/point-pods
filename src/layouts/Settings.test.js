import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Settings from './Settings';

import api from '../api';

import { mockStorage, mockAPI, demockAPI } from '../testData';

beforeEach(() => mockAPI(api, mockStorage));
afterEach(() => demockAPI(api));

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Settings />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', () => {
  const tree = renderer.create(<Settings />);
  expect(tree.toJSON()).toMatchSnapshot();
});
