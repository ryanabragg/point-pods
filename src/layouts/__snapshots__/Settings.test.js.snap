// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `
<div>
  <div
    className=""
  >
    <header
      className="MuiPaper-root-184 MuiPaper-elevation4-190 MuiAppBar-root-176 MuiAppBar-positionFixed-177 MuiAppBar-colorPrimary-182 mui-fixed"
    >
      <div
        className="MuiToolbar-root-211 MuiToolbar-regular-213 MuiToolbar-gutters-212 APIComponent-nowrap-170"
      >
        <button
          aria-label="Back"
          className="MuiButtonBase-root-221 MuiIconButton-root-215 MuiIconButton-colorInherit-216 APIComponent-menuButton-171"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-220"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-224"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-231"
          />
        </button>
        <h2
          className="MuiTypography-root-238 MuiTypography-title-244 MuiTypography-colorInherit-257 APIComponent-flex-169"
        >
          Settings
        </h2>
      </div>
    </header>
    <div
      className="APIComponent-spacer-173"
    />
  </div>
  <div
    className="APIComponent-root-287"
  >
    <div
      className="MuiPaper-root-184 MuiPaper-elevation1-187 MuiPaper-rounded-185 MuiCard-root-291 APIComponent-card-288"
    >
      <div
        className="MuiCardHeader-root-292"
      >
        <div
          className="MuiCardHeader-content-295"
        >
          <span
            className="MuiTypography-root-238 MuiTypography-headline-243 MuiCardHeader-title-296"
          >
            Pod Setup
          </span>
        </div>
      </div>
      <hr
        className="MuiDivider-root-298"
      />
      <div
        className="MuiCardContent-root-302 APIComponent-content-289"
      >
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-303 MuiFormControl-marginDense-305 APIComponent-field-290"
        >
          <label
            className="MuiFormLabel-root-312 MuiInputLabel-root-307 MuiInputLabel-formControl-308 MuiInputLabel-animated-311 MuiInputLabel-marginDense-309"
            data-shrink={false}
            htmlFor="settings-pairingMethod"
          >
            Pairing Method
          </label>
          <div
            className="MuiInput-root-322 MuiInput-formControl-323 MuiInput-underline-326"
          >
            <div
              className="MuiSelect-root-317"
            >
              <div
                aria-haspopup="true"
                aria-owns={null}
                aria-pressed="false"
                className="MuiSelect-select-318 MuiSelect-selectMenu-319 MuiInput-input-330 MuiInput-inputMarginDense-331"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                autoComplete={undefined}
                autoFocus={undefined}
                defaultValue={undefined}
                id="settings-pairingMethod"
                name={undefined}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                rows={undefined}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-224 MuiSelect-icon-321"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-303 MuiFormControl-marginDense-305 APIComponent-field-290"
        >
          <label
            className="MuiFormLabel-root-312 MuiInputLabel-root-307 MuiInputLabel-formControl-308 MuiInputLabel-animated-311 MuiInputLabel-marginDense-309"
            data-shrink={false}
            htmlFor="settings-pairingMethodInitial"
          >
            Initial Pairing Method
          </label>
          <div
            className="MuiInput-root-322 MuiInput-formControl-323 MuiInput-underline-326"
          >
            <div
              className="MuiSelect-root-317"
            >
              <div
                aria-haspopup="true"
                aria-owns={null}
                aria-pressed="false"
                className="MuiSelect-select-318 MuiSelect-selectMenu-319 MuiInput-input-330 MuiInput-inputMarginDense-331"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </div>
              <input
                autoComplete={undefined}
                autoFocus={undefined}
                defaultValue={undefined}
                id="settings-pairingMethodInitial"
                name={undefined}
                onKeyDown={undefined}
                onKeyUp={undefined}
                placeholder={undefined}
                rows={undefined}
                type="hidden"
                value=""
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-224 MuiSelect-icon-321"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-303 MuiFormControl-marginDense-305 APIComponent-field-290"
        >
          <label
            className="MuiFormLabel-root-312 MuiInputLabel-root-307 MuiInputLabel-formControl-308 MuiInputLabel-animated-311 MuiInputLabel-marginDense-309"
            data-shrink={false}
            htmlFor="settings-podSizeMinimum"
          >
            Minimum Pod Size
          </label>
          <div
            className="MuiInput-root-322 MuiInput-formControl-323 MuiInput-underline-326"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-330 MuiInput-inputType-333 MuiInput-inputMarginDense-331"
              defaultValue={undefined}
              disabled={false}
              id="settings-podSizeMinimum"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          aria-describedby={undefined}
          className="MuiFormControl-root-303 MuiFormControl-marginDense-305 APIComponent-field-290"
        >
          <label
            className="MuiFormLabel-root-312 MuiInputLabel-root-307 MuiInputLabel-formControl-308 MuiInputLabel-animated-311 MuiInputLabel-marginDense-309"
            data-shrink={false}
            htmlFor="settings-podSizeMaximum"
          >
            Maximum Pod Size
          </label>
          <div
            className="MuiInput-root-322 MuiInput-formControl-323 MuiInput-underline-326"
          >
            <input
              aria-invalid={false}
              autoComplete={undefined}
              autoFocus={undefined}
              className="MuiInput-input-330 MuiInput-inputType-333 MuiInput-inputMarginDense-331"
              defaultValue={undefined}
              disabled={false}
              id="settings-podSizeMaximum"
              name={undefined}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={undefined}
              onKeyUp={undefined}
              placeholder={undefined}
              readOnly={undefined}
              required={false}
              rows={undefined}
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
